### ISSUE TITLE
Implement login.v1 protocol

### ISSUE THREAD

**Issue by stigok** [nrkno/terraform-registry#20]
https://www.terraform.io/internals/login-protocol

**Comment by stigok** [comment]
I guess #78 should be reverted again when it's time, to ease implementation of this.

**Comment by antonbabenko** [comment]
@stigok Do you have any plans to implement this functionality?

**Comment by stigok** [comment]
@antonbabenko yes, the plans are there, but as usual I'm not sure when I get the time to get around it.

We are looking for an implementation that can handle multiple SSO/SAML providers somewhat generically. The aim is to be able to support GitHub, Google, Azure and more in a simple and straightforward way.

**Comment by stigok** [comment]
Could you please elaborate a bit of your thoughts for this? Might spark some ideas for me too.

**Comment by antonbabenko** [comment]
I thought about just basic implementation of login protocol to be able to login and have a private Terraform module registry experience. Nothing fancy.
